File: py/uniphant/parse_arguments.py
from argparse import ArgumentParser
from sys import stderr, exit
from typing import Tuple, Optional
from uuid import UUID
from .utils import is_valid_uuid

def parse_arguments() -> Tuple[Optional[UUID], bool]:
    parser = ArgumentParser()
    parser.add_argument('worker_id',
                        nargs='?',
                        default=None,
                        help='UUID')
    parser.add_argument('-d', '--daemonize',
                        action='store_true',
                        default=False,
                        help='run as a forking daemon')
    args = parser.parse_args()
    worker_id: UUID = None
    daemonize: bool = args.daemonize
    if args.worker_id is not None:
        if not is_valid_uuid(args.worker_id):
            print(f"The specified worker_id {args.worker_id} is not a valid UUID")
            parser.print_usage(stderr)
            exit(2)
        worker_id = UUID(args.worker_id)
    return worker_id, daemonize

File: rust/src/parse_arguments.rs
use std::env;
use std::process;
use uuid::Uuid;
use structopt::StructOpt;

#[derive(StructOpt, Debug)]
#[structopt(name = "app")]
pub struct Args {
    /// UUID
    #[structopt(short, long, required = false, parse(try_from_str = parse_uuid))]
    worker_id: Option<Uuid>,

    /// Run as a forking daemon
    #[structopt(short = "d", long = "daemonize")]
    daemonize: bool,
}

fn parse_uuid(src: &str) -> Result<Uuid, &'static str> {
    if utils::is_valid_uuid(src) {
        Ok(Uuid::parse_str(src).unwrap())
    } else {
        Err("The specified worker_id is not a valid UUID")
    }
}

pub fn parse_arguments() -> Args {
    let args: Args = Args::from_args();
    args
}

fn main() {
    let (worker_id, daemonize) = {
        let args = parse_arguments();
        (args.worker_id, args.daemonize)
    };

    // Your application logic here
}
